nemo/collections/tts/models/magpietts.py

```diff
@@ -176,6 +178,20 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
         self.pad_context_text_to_max_duration = self.model_type in ['decoder_context_tts', 'decoder_ce']
         self.use_kv_cache_for_inference = cfg.get('use_kv_cache_for_inference', False)
 
+        # Below args (text_context_remapping_json, text_context_remapping_prob) are
+        # for combining multiple context_texts into a single one during training.
+        # Eg. if we want to treat Emma_neutral and Emma_conversational as one speaker,
+        # we can create an override dict {'Emma_neutral' : 'Emma', 'Emma_conversational' : 'Emma'}
+        # This dict is saved in a json file given by cfg.model.text_context_remapping_json
+        # If we want to preserve both behaviours i.e (Emma_neutral, Emma_conversational) and just (Emma)
+        # we can do this mapping with a probability during training, as specified by text_context_remapping_prob
+        self.text_context_remapping = None
+        text_context_remapping_json = cfg.get('text_context_remapping_json', None)
+        self.text_context_remapping_prob = cfg.get('text_context_remapping_prob', 0.0)
+        if text_context_remapping_json is not None:
+            with open(text_context_remapping_json, 'r') as f:
+                self.text_context_remapping = json.load(f)
```

Comment:
Why do we want to pass it in as a json file rather than saving the dictionary inside the yaml itself? This seems like an anti-design pattern.

Reply:
We might want nested yamls in that case. Inside the same yaml (or in the training command), it can become very huge, Eg. I had 60 entries.